### PR TITLE
[Order editing] Non-default currency order edit notice

### LINKED ISSUE #14304: Editing Order Details - Incorrect Currency Displayed When Order Currency Differs from Store Default
We consistently use the store's default currency when editing an order, even if the order was created in a different currency. This results in inconsistencies across various sections of the editable order screen, including Items, Custom Amounts, Discounts, Coupons, and Order Totals. Additionally, when adding new products to the order, prices are displayed in the store's default currency instead of the order's currency.

This inconsistency is confusing for users.

https://github.com/user-attachments/assets/abdc1217-476c-449e-abd6-68b06ee3e1fb

Suggested Workaround to consider:
> To completely fix this issue, we would need to receive product prices in the order’s currency so we can display them correctly during product selection. Since the REST API does not currently provide this, I suggest disabling the Edit mode for orders in a currency different from the store’s default.
>
> Users would still be able to change the order status, create shipping labels, issue refunds, and add tracking, notes, and shipping details from the Order Details. They wouldn’t be able to add Custom Amounts, Discounts, Coupons, or new Items to the order. These actions aren’t very common for orders created on the storefront anyways.

peaMlT-XF-p2

### PR DESCRIPTION
**Pull request by joshheald**
Closes: #14304 

## Description

This PR disables order editing for orders with a different currency compared to the site's default.

This is a workaround. If we allowed this, items on these orders would be shown in the wrong currency. See peaMlT-XF-p2 for context.

Multi-currency is not supported in the app at present, but when we add support properly, this workaround should be removed and the edit screen fixed.

## Steps to reproduce

1. Launch the app and switch to a multi currency store
2. Go to the orders tab and select an order in a different currency from the store's default
3. Tap edit
4. Observe that you're shown a notice explaining that you can't edit in the app

Repeat for an order in your store's default currency – observe that you can edit it as normal.

## Testing information

I added some unit tests for this, but the reliance on whether the order is synced or not was tricky. I effectively exposed that property to be mockable in tests, which is a little strange but it would have made for very brittle tests to actually allow the `syncEverything` function to work via lots of mocking.

I tested on an iPhone running iOS 18.1, and an iPad running iOS 17.7

Tracks event registered.

Note that I also see issues when creating an order – the order creation flow shows my site's default currency (GBP), but the order is then created in USD. I'm not really sure why that is, and it seems more drastic to disable order creation than order editing... so I've treated it as out of scope for this ticket, but will find out why and create another issue if needed.

## Screenshots

https://github.com/user-attachments/assets/4c4ec638-0ef0-4187-b561-eb327bbebc93

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by jaclync** [comment]
👋 Sorry I saw the PR late, I will review it within a few hours.

File: WooCommerce/Classes/Analytics/WooAnalyticsEvent.swift [hunk 1]
```diff
@@ -654,6 +654,10 @@ extension WooAnalyticsEvent {
             ])
         }
 
+        static func orderEditButtonTappedWhileDisabledForCurrencyConflict() -> WooAnalyticsEvent {
+            WooAnalyticsEvent(statName: .orderEditButtonTappedWhileDisabledForCurrencyConflict, properties: [:])
```
Comment (jaclync) [inline WooCommerce/Classes/Analytics/WooAnalyticsEvent.swift:658]: Thanks for adding a Tracks event to see how many merchants are encountering this to have a better idea about the priority 👍

**Review by jaclync — APPROVED** [review]
LGTM! Works as expected :shipit: 



> Note that I also see issues when creating an order – the order creation flow shows my site's default currency (GBP), but the order is then created in USD. I'm not really sure why that is, and it seems more drastic to disable order creation than order editing... so I've treated it as out of scope for this ticket, but will find out why and create another issue if needed.

Is this happening with some multi-currency plugin or setting? I was testing with a typical store with previous orders created with a different currency and I can't reproduce this.

**Comment by joshheald** [comment]
> Is this happening with some multi-currency plugin or setting? I was testing with a typical store with previous orders created with a different currency and I can't reproduce this.

No, it happens because order creation uses the logged in user's default currency as the one for the order when it's not specified in the request.

To change that from the site default, you have to go to `/my-account/edit-account/`, which I had done and forgotten about on this test store/user.

I'm just checking whether there are implications on that for editing as well; this PR may need to change or be followed up if it does. It's been bumped to 21.2 now anyway so it's fine.

**Comment by joshheald** [comment]
Oh and thanks for the review @jaclync 😄

**Comment by joshheald** [comment]
> Is this happening with some multi-currency plugin or setting?

Actually yes... I doubt the user-level currency setting is available without a plugin.

My test site uses the multi-currency features built in to WooPayments.

**Pull request merged**